src/rpc.js

```diff
@@ -152,15 +152,21 @@ class RPC {
     if (!this.socket || this.socket.readyState !== this.socket.OPEN) {
       return this.Promise.reject(new Error('Not connected'));
     }
-    this.requestId += 1;
+    if (!data.id) {
+      data.id = this.createRequestId();
+    }
     data.jsonrpc = '2.0';
-    data.id = this.requestId;
     return new this.Promise((resolve, reject) => {
       this.socket.send(JSON.stringify(data));
       this.emit('traffic', 'sent', data);
       return this.registerResolver(data.id, resolve, reject);
     });
   }
+
+  createRequestId() {
```

Comment:
We had to move this logic to enable the session to assign the Id since the initial promise returned from `session.send()` needs to have the `requestId` exposed.